sigopt/hyperopt/base.py

```diff
@@ -38,35 +41,32 @@ def upload(self, trials=None):
     self.uploaded_tids.update(ids)
     return ids
 
-  def _upload(self, trials):
-    ids = {}
-    for trial in trials:
-      tid = trial['tid']
-      result = trial['result']
-      metrics = {k:v for k, v in result.items() if isinstance(v, (int, float))}
-      parameters = self.trial_parameters(trial)
-      metadata = {'optimizer': 'hyperopt'}
-      status = result.get('status')
-      run = self.factory.create_run(metadata=metadata)
-      with run:
-        self.log_run_params(run, parameters)
-        if status == STATUS_OK:
-          run.log_metrics(metrics)
-        elif status == STATUS_FAIL:
-          run.log_failure()
-        else:
-          raise ValueError(f'status must be {STATUS_OK} or {STATUS_FAIL}, actully {status}')
-      ids[tid] = run.id
-    return ids
+  def trial_to_run(self, trial):
+    metadata = {'optimizer': 'hyperopt'}
+    result = trial['result']
```

Comment:
let's make sure that `trial` is a `trial` by adding: 
```self.assert_valid_trial(trial)```

Reply:
It is low priority as by now trial is create by Trials, all are valid. If user manually created some invalid trials, the upload will be failed, which is acceptable.

Reply:
oh, I'm sorry, I didn't mean to say that you have to implement anything new just add 
`self.assert_valid_trial(trial)` or more explicitly `self._trials.assert_valid_trial(trial)` before line 45. That will be more than enough to guarantee that we always have the fields that we use for any custom implementation of the HyperOpt `Trials` class.

If you think that the likelihood of someone implementing a custom `Trials` class without the required fields is minimal, then we can just check if the `trials` object is a a subclass of `Trials` in the `__init__` of `SigOptTrials`; I like that.

Reply:
Validation added. In `upload(trials)`, the trials do not need to be `Trials`, it requires an iterator of trial.

Reply:
hm... I think you are using the fact that `_trials` are from `Trials`, for example, when you make the wrapper of the `refresh` method, might as well use the already implemented method `_trials.assert_valid_trial(trial)`.

happy to reconsider if you see a problem with that, but I tested it locally and it seems to work

Reply:
Change with `_trials.assert_valid_trial(trial)`.